src/Microsoft.AspNet.Mvc.Core/MvcRouteHandler.cs

```diff
@@ -102,14 +108,23 @@ public VirtualPathData GetVirtualPath([NotNull] VirtualPathContext context)
             var invoker = invokerFactory.CreateInvoker(actionContext);
             if (invoker == null)
             {
-                LogActionSelection(actionSelected: true, actionInvoked: false, handled: context.IsHandled);
-
                 throw new InvalidOperationException(
                     Resources.FormatActionInvokerFactory_CouldNotCreateInvoker(
                         actionDescriptor.DisplayName));
             }
 
-            await invoker.InvokeAsync();
+            var controllerActionDescriptor = actionDescriptor as ControllerActionDescriptor;
+            var controllerName = string.Empty;
+            if (controllerActionDescriptor != null)
+            {
+                controllerName = controllerActionDescriptor.ControllerTypeInfo.FullName;
+            }
+
+            using (_logger.BeginScope("Executing action '{ActionName}' on controller '{ControllerName}'.", 
+                actionDescriptor.Name, controllerName))
```

Comment:
Please refer to the code sample I sent via email and do **exactly that**. We spent a good deal of time discussing this exact piece of code.